scripts/verified-catalog-e2e-common.sh

```diff
@@ -0,0 +1,301 @@
+            done
+
+            if [[ -z ${all_status} || -z ${reason} ]];then
+                echo "Could not find a created taskrun or pipelinerun in ${target_ns}"
+            fi
+
+            breakit=True
+            for status in ${all_status};do
+                [[ ${status} == *ERROR || ${reason} == *Fail* || ${reason} == Couldnt* ]] && show_failure ${testname} ${target_ns}
+                if [[ ${status} != True ]];then
+                    breakit=
+                fi
+            done
+
+            if [[ ${breakit} == True ]];then
+                unset task_to_wait_for[$testname]
+                [[ -z ${CATALOG_TEST_SKIP_CLEANUP} ]] && ${KUBECTL_CMD} delete ns ${target_ns} >/dev/null
+                echo "${started}::$(date '+%Hh%M:%S') SUCCESS: ${testname} testrun has successfully executed" ;
+            fi
+        done
+
+        sleep 10
+        cnt=$((cnt+1))
+    done
+    set -x 
+}
+
+function show_failure() {
+    local testname=$1 tns=$2
+
+    echo "FAILED: ${testname} task has failed to comeback properly" ;
+    echo "--- Task Dump"
+    ${KUBECTL_CMD} get -n ${tns} task -o yaml
+    echo "--- Pipeline Dump"
+    ${KUBECTL_CMD} get -n ${tns} pipeline -o yaml
+    echo "--- PipelineRun Dump"
+    ${KUBECTL_CMD} get -n ${tns} pipelinerun -o yaml
+    echo "--- TaskRun Dump"
+    ${KUBECTL_CMD} get -n ${tns} taskrun -o yaml
+    echo "--- Container Logs"
+    for pod in $(${KUBECTL_CMD} get pod -o name -n ${tns}); do
+        ${KUBECTL_CMD} logs --all-containers -n ${tns} ${pod} || true
+    done
+    exit 1
+}
+
+function install_pipeline_crd() {
+  local latestreleaseyaml
+  echo ">> Deploying Tekton Pipelines"
+  if [[ -n ${RELEASE_YAML} ]];then
+	latestreleaseyaml=${RELEASE_YAML}
+  else
+    latestreleaseyaml="https://storage.googleapis.com/tekton-releases/pipeline/latest/release.yaml"
+  fi
+  [[ -z ${latestreleaseyaml} ]] && fail_test "Could not get latest released release.yaml"
+  ${KUBECTL_CMD} apply -f ${latestreleaseyaml} ||
+      fail_test "Build pipeline installation failed"
+
+  # Make sure that eveything is cleaned up in the current namespace.
+  for res in pipelineresources tasks pipelines taskruns pipelineruns; do
+    ${KUBECTL_CMD} delete --ignore-not-found=true ${res}.tekton.dev --all
+  done
+
+  # Wait for pods to be running in the namespaces we are deploying to
+  wait_until_pods_running tekton-pipelines || fail_test "Tekton Pipeline did not come up"
+}
+
+function test_tasks {
+    local cnt=0
+    local task_to_tests=""
+
+    for runtest in $@;do
+        task_to_tests="${task_to_tests} ${runtest}"
+        if [[ ${cnt} == "${MAX_NUMBERS_OF_PARALLEL_TASKS}" ]];then
+            test_task_creation "${task_to_tests}"
+            cnt=0
+            task_to_tests=""
+            continue
+        fi
+        cnt=$((cnt+1))
+    done
+
+    # in case if there are some remaining tasks
+    if [[ -n ${task_to_tests} ]];then
+        test_task_creation "${task_to_tests}"
```

Comment:
I guess we don't need this look now. This was required when no of tasks within catalog were increasing at a good amount of rate but now since we are moving to decentralised catalogs, we don't need this

Reply:
In nightly test we will test all the versions/releases of the catalog, which could scale to >8 tests (in the future), so I was thinking to keep this so that we can have some safety limit in the test when we have multiple versions in the future? 🤔

Reply:
but in that case, should we not just test the latest 3 versions of the applications?

Reply:
In the current design of TEP-0079 and TEP-0115, I don't think we called out that we will only support 3 latest versions of the catalog, so I think we should test all of them. But we can definitely further discuss this option later.